probe_src/python/probe_py/manual/cli.py

```diff
@@ -146,5 +195,539 @@ def dump(
+        typer.echo("File transfer successful.")
+    except subprocess.CalledProcessError as e:
+        # Capture and print the error message
+        typer.echo(f"Error occurred during file transfer: {e}")
+        typer.echo(f"Exit code: {e.returncode}")
+        typer.echo(f"Error output: {e.stderr}")
+
+
+def check_and_create_remote_file(hostname:str, port:int, username:str, remote_file_path:pathlib.Path) -> None:
+    try:
+        # Check if the file exists on the remote server
+        check_command = f"ssh -p {port} {username}@{hostname} 'test -f {remote_file_path} && echo \"File exists\" || echo \"File does not exist\"'"
+        result = subprocess.run(check_command, shell=True, check=True, capture_output=True, text=True)
+        output = result.stdout.strip()
+        if output == "File does not exist":
+            # Create the file if it does not exist
+            create_command = f"ssh -p {port} {username}@{hostname} 'touch {remote_file_path}'"
+            subprocess.run(create_command, shell=True, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed with exit code {e.returncode}")
+        print(f"Error output: {e.stderr}")
+    except Exception as e:
+        print(f"An error occurred: {e}")
+
+
+def get_remote_home(remote_user:str, remote_host: str, port: int)->pathlib.Path:
+    remote_xdg = get_remote_xdg_data_home(remote_user, remote_host, port)
+    if remote_xdg == "":
+        home = get_remote_home_env(remote_user, remote_host, port)
+        remote_home = pathlib.Path(f"{home}/.local/share") / "PROBE"
+    else:
+        remote_home = pathlib.Path(f"home/{remote_xdg}/.local/share") / "PROBE"
+    return remote_home
+
+def get_remote_xdg_data_home(remote_user: str, remote_host: str, port: int)-> str | None:
+    try:
+        ssh_command = [
+            "ssh",
+            "-p",
+            str(port),  # Specify the port
+            f"{remote_user}@{remote_host}",  # Remote user and host
+            "echo $XDG_DATA_HOME",  # Command to retrieve XDG_DATA_HOME
+        ]
+        result = subprocess.run(ssh_command, capture_output=True, text=True, check=True)
+        xdg_data_home = result.stdout.strip()
+        return xdg_data_home
+    except subprocess.CalledProcessError as e:
+        print(f"Error retrieving XDG_DATA_HOME: {e}")
+        return None
+
+def get_remote_home_env(remote_user: str, remote_host: str, port: int)->str | None:
+    try:
+        ssh_command = [
+            "ssh",
+            "-p",
+            str(port),  # Specify the port
+            f"{remote_user}@{remote_host}",  # Remote user and host
+            "echo $HOME",  # Command to retrieve HOME
+        ]
+        result = subprocess.run(ssh_command, capture_output=True, text=True, check=True)
+        home = result.stdout.strip()
+        return home
+    except subprocess.CalledProcessError as e:
+        print(f"Error retrieving XDG_DATA_HOME: {e}")
+        return None
```

Comment:
In such cases, it is customary to return `/homeless-shelter`.
This isn't a joke, or rather it isn't *my* joke: If the user doesn't have a `$HOME`, then they are probably a single-user, so they are probably root. If they are root, any old directory will work fine, as long as it is persistent and they have permissions on it (which should be if they have root).

But it's an edge case that rarely comes up. Mostly, it's much nicer to have code that always returns as string (no need to check the result) than one that sometimes returns `None`, no matter how rare.